jest.config.mjs

```diff
@@ -0,0 +1,36 @@
+// /*
+//  * For a detailed explanation regarding each configuration property, visit:
+//  * https://jestjs.io/docs/configuration
+//  */
+
+/** @type {import('ts-jest/dist/types').InitialOptionsTsJest} */
+export default {
+    projects: [
+        {
+            displayName: 'ESLint',
+            runner: 'jest-runner-eslint',
+            watchPlugins: ['jest-runner-eslint/watch-fix'],
+        },
+        {
+            displayName: 'ts-check and test',
+            extensionsToTreatAsEsm: ['.ts'],
+            preset: 'ts-jest/presets/default-esm',
+            moduleNameMapper: {
+                '^(\\.{1,2}/.*)\\.js$': '$1',
+            },
+            globals: {'ts-jest': {tsconfig: '<rootDir>/../tsconfig.test.json', useESM: true}},
+            transform: {
+                '^.+\\.[tj]s$': ['ts-jest'],
+            },
+            clearMocks: true,
+            collectCoverage: true,
+            collectCoverageFrom: ['<rootDir>/**/*.{ts,tsx}'],
+            coverageDirectory: 'coverage',
+            setupFiles: ['../jest.setup.ts'],
+            // Compile/transform the only prod dep that is not mocked in our tests, query-string, and its own prod deps.
+            transformIgnorePatterns: ['node_modules/(?!(query-string|decode-uri-component|split-on-first|filter-obj))'],
```

Comment:
Since this config is a javascript file, I think it may be worth extracting the list of dependencies to ignore in a separate variable. That would also remove the need for the comment.

E.g.:
```js
const unmockedDependencies = [
    'query-string',
    'decode-uri-component',
    'split-on-first',
    'filter-obj',
];

export default {
    ...
    {
        ...,
        transformIgnorePatterns: [`node_modules/(?!(${unmockedDependencies.join('|')}))`],
    }
}
```

Reply:
That's a good idea 👍